File: app/src/main/java/yoyo/app/android/com/yoyoapp/trip/schedule/ScheduleViewModel.java
package yoyo.app.android.com.yoyoapp.trip.schedule;

import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import yoyo.app.android.com.yoyoapp.DataModels.Schedule;

import java.util.ArrayList;

public class ScheduleViewModel extends AndroidViewModel {

    private MutableLiveData<ArrayList<Schedule>> scheduleMutableLiveData;
    private ScheduleRepository scheduleRepository;

    public ScheduleViewModel(@NonNull Application application) {
        super(application);
        scheduleRepository = ScheduleRepository.getInstance(getApplication());
    }

    public void initSchedule(String tripId, long startDate, long endDate)
    {
        scheduleMutableLiveData = new MutableLiveData<>();
        scheduleMutableLiveData = scheduleRepository.getSchedule(tripId,startDate,endDate);
    }

    public LiveData<ArrayList<Schedule>> getScheduleList()
    {
        return scheduleMutableLiveData;
    }


}

File: app/src/main/java/yoyo/app/android/com/yoyoapp/trip/schedule/ScheduleViewModel.kt
package yoyo.app.android.com.yoyoapp.trip.schedule

import android.app.Application
import androidx.core.util.Consumer
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import yoyo.app.android.com.yoyoapp.DataModels.Schedule

import java.util.ArrayList

class ScheduleViewModel(application: Application) : AndroidViewModel(application) {

    private var scheduleMutableLiveData: MutableLiveData<ArrayList<Schedule>>? = null
    private val scheduleRepository: ScheduleRepository = ScheduleRepository(getApplication())

    val scheduleList: LiveData<ArrayList<Schedule>>?
        get() = scheduleMutableLiveData

    fun initSchedule(tripId: String, startDate: Long, endDate: Long) {
        scheduleMutableLiveData = MutableLiveData()
        scheduleRepository.getSchedule(tripId, startDate, endDate, Consumer {
            scheduleMutableLiveData?.value = it
        })
    }


}
